exploratory/app.R

```diff
@@ -1,62 +1,61 @@
-options("teal.ggplot2_args" = teal.widgets::ggplot2_args(labs = list(caption = "NEST PROJECT")))
-
 library(teal.modules.general)
-library(scda)
-library(scda.2022)
-library(dplyr)
-library(tidyr)
-library(nestcolor)
-# optional libraries
-library(broom)
-library(colourpicker)
-library(ggExtra)
-library(ggpmisc)
-library(ggpp)
-library(goftest)
-library(gridExtra)
-library(htmlwidgets)
-library(jsonlite)
-library(lattice)
-library(MASS)
-library(rlang)
-library(rtables)
-library(sparkline)
-
-options(shiny.useragg = FALSE)
+options(
+  shiny.useragg = FALSE,
+  teal.ggplot2_args = teal.widgets::ggplot2_args(labs = list(caption = "NEST PROJECT"))
+)
 
 nest_logo <- "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png"
 
-# code>
-ADSL <- synthetic_cdisc_data("latest")$adsl
-ADRS <- synthetic_cdisc_data("latest")$adrs
-ADLB <- synthetic_cdisc_data("latest")$adlb
+## Data reproducible code ----
+data <- teal_data()
+data <- within(data, {
+  library(scda)
+  library(scda.2022)
+  library(dplyr)
+  library(tidyr)
+  library(ggExtra)
+  library(ggpmisc)
+  library(ggpp)
+  library(goftest)
+  library(gridExtra)
+  library(htmlwidgets)
+  library(jsonlite)
+  library(lattice)
+  library(MASS)
+  library(rlang)
+  library(rtables)
+  library(nestcolor)
+  library(broom)
+  library(colourpicker)
+  library(sparkline)
 
-ADLBPCA <- ADLB %>%
-  dplyr::select(USUBJID, STUDYID, SEX, ARMCD, AVAL, AVISIT, PARAMCD) %>%
-  tidyr::pivot_wider(
-    values_from = "AVAL",
-    names_from = c("PARAMCD", "AVISIT"),
-    names_sep = " - "
-  )
 
-adsl <- cdisc_dataset("ADSL", ADSL, code = "ADSL <- synthetic_cdisc_data(\"latest\")$adsl")
-adrs <- cdisc_dataset("ADRS", ADRS, code = "ADRS <- synthetic_cdisc_data(\"latest\")$adrs")
-adlb <- cdisc_dataset("ADLB", ADLB, code = "ADLB <- synthetic_cdisc_data(\"latest\")$adlb")
-adlbpca <- cdisc_dataset(
-  "ADLBPCA",
-  ADLBPCA,
-  code = 'ADLBPCA <- ADLB %>%
+  ADSL <- synthetic_cdisc_data("latest")$adsl
+  ADRS <- synthetic_cdisc_data("latest")$adrs
+  ADLB <- synthetic_cdisc_data("latest")$adlb
+
+  ADLBPCA <- ADLB %>%
     dplyr::select(USUBJID, STUDYID, SEX, ARMCD, AVAL, AVISIT, PARAMCD) %>%
     tidyr::pivot_wider(
       values_from = "AVAL",
       names_from = c("PARAMCD", "AVISIT"),
       names_sep = " - "
-    )',
-  keys = c("STUDYID", "USUBJID"),
-  label = "ADLB reshaped",
-  vars = list(ADLB = adlb)
-)
-# <code
+    )
+})
+
+datanames <- c("ADSL", "ADRS", "ADLB", "ADLBPCA")
+datanames(data) <- datanames
+
+join_keys(data) <- default_cdisc_join_keys[datanames]
```

Comment:
should we exclude `ADSLPCA` since it is not present in the default_cdisc_join_keys. Instead, we should simply use c("ADSL", "ADRS", "ADLB").to enhance clarity in the subsequent step ?

Reply:
Makes sense to `default_cdisc_join_keys[c("ADSL", "ADRS", "ADLB")] # no ADLBPCA defaults`

Reply:
I think it's just a matter of DRY here. We already store the datanames in a variable so it's convenient to do a subset like this. Now it's okay, previously we used to get a join key set for `ADLB` with NULL.